File: client/src/api/music.api.ts
import { api } from './client.js';
import { NexaMusicTrack } from '../types/music.types.js';

/**
 * Fetch tracks from backend.
 * @param query Optional search keyword
 * @param genre Optional genre filter
 * @returns List of NexaMusicTrack
 */
export async function searchJamendoTracks(query?: string, genre?: string): Promise<NexaMusicTrack[]> {
  const params: Record<string, string> = {};
  let endpoint = '/music/tracks';

  if (query) {
    endpoint = '/music/search';
    params.q = query;
  } else if (genre) {
    endpoint = `/music/genres/${encodeURIComponent(genre)}`;
  }

  const resp = await api.get(endpoint, { params });
  return resp.data.data as NexaMusicTrack[];
}

export async function getTrackById(trackId: string): Promise<NexaMusicTrack> {
  const resp = await api.get(`/music/tracks/${encodeURIComponent(trackId)}`);
  return resp.data.data as NexaMusicTrack;
}

File: android/app/src/main/java/com/nexa/social/data/api/MusicApi.kt
package com.nexa.social.data.api

import com.nexa.social.data.models.ApiResponse
import com.nexa.social.data.models.MusicTrack
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface MusicApi {
    @GET("music/tracks")
    suspend fun getTracks(
        @Query("limit") limit: Int = 20
    ): Response<ApiResponse<List<MusicTrack>>>

    @GET("music/search")
    suspend fun searchTracks(
        @Query("q") query: String,
        @Query("limit") limit: Int = 20
    ): Response<ApiResponse<List<MusicTrack>>>

    @GET("music/genres/{genre}")
    suspend fun getTracksByGenre(
        @Path("genre") genre: String
    ): Response<ApiResponse<List<MusicTrack>>>

    @GET("music/tracks/{trackId}")
    suspend fun getTrackById(
        @Path("trackId") trackId: String
    ): Response<ApiResponse<MusicTrack>>
}
